packages/flutter/test/widgets/selectable_region_test.dart

```diff
@@ -886,6 +886,162 @@ void main() {
+            ),
+          ),
+        );
+
+        expect(buttonTypes.isEmpty, true);
+        expect(find.byKey(toolbarKey), findsNothing);
+
+        final RenderParagraph paragraph = tester.renderObject<RenderParagraph>(find.descendant(of: find.text('How are you?'), matching: find.byType(RichText)));
+        final TestGesture gesture = await tester.startGesture(textOffsetToPosition(paragraph, 2));
+        addTearDown(gesture.removePointer);
+        await tester.pump(const Duration(milliseconds: 500));
+        await tester.pumpAndSettle();
+
+        // Should show the selection handles when the long press starts.
+        List<FadeTransition> transitions = find.descendant(
+          of: find.byWidgetPredicate((Widget w) => '${w.runtimeType}' == '_SelectionHandleOverlay'),
+          matching: find.byType(FadeTransition),
+        ).evaluate().map((Element e) => e.widget).cast<FadeTransition>().toList();
+        expect(transitions.length, 2);
+        FadeTransition left = transitions[0];
+        FadeTransition right = transitions[1];
+        expect(paragraph.selections[0], const TextSelection(baseOffset: 0, extentOffset: 3));
+        expect(left.opacity.value, equals(1.0));
+        expect(right.opacity.value, equals(1.0));
+        expect(find.byKey(toolbarKey), findsNothing);
+
+        await gesture.moveTo(textOffsetToPosition(paragraph, 8));
+        await tester.pumpAndSettle();
+        transitions = find.descendant(
+          of: find.byWidgetPredicate((Widget w) => '${w.runtimeType}' == '_SelectionHandleOverlay'),
+          matching: find.byType(FadeTransition),
+        ).evaluate().map((Element e) => e.widget).cast<FadeTransition>().toList();
+        expect(transitions.length, 2);
+        left = transitions[0];
+        right = transitions[1];
+
+        // Should show the selection handles while doing a long press drag.
+        expect(paragraph.selections[0], const TextSelection(baseOffset: 0, extentOffset: 11));
+        expect(left.opacity.value, equals(1.0));
+        expect(right.opacity.value, equals(1.0));
+        expect(find.byKey(toolbarKey), findsNothing);
+
+        await gesture.up();
+        await tester.pumpAndSettle();
+        transitions = find.descendant(
+          of: find.byWidgetPredicate((Widget w) => '${w.runtimeType}' == '_SelectionHandleOverlay'),
+          matching: find.byType(FadeTransition),
+        ).evaluate().map((Element e) => e.widget).cast<FadeTransition>().toList();
+        expect(transitions.length, 2);
+        left = transitions[0];
+        right = transitions[1];
+
+        // Should show the selection handles and context menu when long press ends.
+        expect(paragraph.selections[0], const TextSelection(baseOffset: 0, extentOffset: 11));
+        expect(left.opacity.value, equals(1.0));
+        expect(right.opacity.value, equals(1.0));
+        expect(find.byKey(toolbarKey), findsOneWidget);
+      },
+      variant: TargetPlatformVariant.only(TargetPlatform.iOS),
```

Comment:
is it possible to combine these two test and conditionally check the selection handles show up according to platform?